cmake/BundledSpdlog.cmake

```diff
@@ -0,0 +1,74 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+function(use_bundled_spdlog SOURCE_DIR BINARY_DIR)
+    # Define byproducts
+    if (WIN32)
+        if ("${CMAKE_BUILD_TYPE}" STREQUAL "Debug")
+            set(BYPRODUCT "lib/spdlogd.lib")
+        else()
+            set(BYPRODUCT "lib/spdlog.lib")
+        endif()
+    else()
+        include(GNUInstallDirs)
+        if ("${CMAKE_BUILD_TYPE}" STREQUAL "Debug")
+            set(BYPRODUCT "${CMAKE_INSTALL_LIBDIR}/libspdlogd.a")
+        else()
+            set(BYPRODUCT "${CMAKE_INSTALL_LIBDIR}/libspdlog.a")
+        endif()
```

Comment:
What happened here? I think I remember a commit which changed these to generator expressions, but now the generator expressions are not there.

Reply:
https://gitlab.kitware.com/cmake/cmake/-/issues/19522